### PR TITLE
Improvements to the schema for the key_value processor.

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

This adds a none option to the whitespace enum while still supporting empty string. It includes default values to prevent fields from displaying that they are required. It also provides a solution for the schema generation which removes any property with a defaultValue from the list of required fields.

 
### Issues Resolved

N/A
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/module/DataPrepperJakartaValidationModule.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class DataPrepperJakartaValidationModule extends JakartaValidationModule {
+    public DataPrepperJakartaValidationModule(final JakartaValidationOption... options) {
+        super(options);
+    }
+
+    @Override
+    protected boolean isRequired(final MemberScope<?, ?> member) {
+        final JsonProperty jsonPropertyAnnotation = member.getAnnotationConsideringFieldAndGetter(JsonProperty.class);
+        if(jsonPropertyAnnotation != null) {
+            if(jsonPropertyAnnotation.defaultValue() != null && !jsonPropertyAnnotation.defaultValue().isEmpty()) {
```
Comment (chenqi0805) [inline data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/module/DataPrepperJakartaValidationModule.java:27]: Do we have any cases where null value is allowed but defaultValue is not null?

File: data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/TransformOption.java [hunk 2]
```diff
@@ -44,6 +45,8 @@ Function<String, String> getTransformFunction() {
 
     @JsonCreator
     public static TransformOption fromTransformName(final String transformName) {
+        if(Objects.equals(transformName, ""))
```
Comment (chenqi0805) [inline data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/TransformOption.java:48]: Nit: space

File: data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/module/DataPrepperJakartaValidationModule.java (hunk 1 above)
Comment (dlvenable) [reply data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/module/DataPrepperJakartaValidationModule.java:27]: Technically you cannot provide `null` to the the annotation's `defaultValue` field. So this should never be reached. I added the `!= null` as a defensive programming approach.

Comment (dlvenable) [reply data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/module/DataPrepperJakartaValidationModule.java]: For a more general point-of-view, anytime that a `defaultValue` is provided, regardless of the value, then the field is not required.

Comment (chenqi0805) [reply data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/module/DataPrepperJakartaValidationModule.java]: Sorry my misread. I thought this condition means required.

**Review by chenqi0805 — APPROVED** [review]

**Review by graytaylor0 — APPROVED** [review]

**Pull request merged**